core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1965,6 +1965,13 @@ private boolean needNewSubQuery(
         return true;
       }
 
+      if (rel instanceof Aggregate
```

Comment:
seems you need an extra check for null here

Reply:
Done, the wrong reference object was used here.